sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -18,8 +18,9 @@
 #include <atomic>
 #include <cassert>
 #include <iostream>
+#include <limits>
 #include <mutex>
-#include <unordered_map>
+#include <stack>
```

Comment:
As I recall, we talked about copying L0 GPU plugin behavior here, which you said used unordered_map and did not care about order of insertion/removal, so that we don't introduce unexpected failures - ?
@romanovvlad, do you agree with that?

Reply:
I wouldn't expect it introducing new failures. Also, `std::stack` should be faster. Do not mind if we use `std::unordered_map` and have a TODO suggesting replacement if there are concerns.

Reply:
@dongkyunahn-intel will revert to unordered_map and leave TODO comments for changing to stack.

Reply:
Done.